server/src/main/java/org/elasticsearch/action/admin/cluster/configuration/AddVotingConfigExclusionsRequest.java

```diff
@@ -72,16 +74,22 @@ public AddVotingConfigExclusionsRequest(StreamInput in) throws IOException {
 
     Set<VotingConfigExclusion> resolveVotingConfigExclusions(ClusterState currentState) {
         final DiscoveryNodes allNodes = currentState.nodes();
-        final Set<VotingConfigExclusion> resolvedNodes = Arrays.stream(allNodes.resolveNodes(nodeDescriptions))
+        final DiscoveryNodes.NodeResolutionResults nodeResolutionResults = allNodes.resolveNodesExact(nodeDescriptions);
```

Comment:
I think this should be stricter about interpreting the node descriptions differently depending on their source (whether old-style deprecated URI or one of the two new-style parameters). This will entail adding more fields to this request class, and supporting backwards-compatibility on (de)serialisation by behaving differently according to `StreamOutput#getVersion` and `StreamInput#getVersion`.

As it is, this is too lenient.

Reply:
To be more precise, I think `AddVotingConfigExclusionsRequest` should have three `String[]` fields, one for old-style `nodeDescriptions`, one for `nodeIds` and one for `nodeNames`, no more than one of which may be nonempty.

Reply:
Done. However, I'm not sure which version to use to enable this feature, do I need to create a new version? I've put TODOs in the relevant section of the code to highlight this.